python-package/lightgbm/basic.py

```diff
@@ -41,20 +41,26 @@ def info(self, msg: str) -> None:
     def warning(self, msg: str) -> None:
         warnings.warn(msg, stacklevel=3)
 
+    def error(self, msg: str) -> None:
+        warnings.warn(msg, stacklevel=3)
+
 
-_LOGGER: Union[_DummyLogger, Logger] = _DummyLogger()
+_LOGGER: Any = _DummyLogger()
 
 
-def register_logger(logger: Logger) -> None:
+def register_logger(logger: Any) -> None:
     """Register custom logger.
 
     Parameters
     ----------
-    logger : logging.Logger
+    logger : Any
         Custom logger.
     """
-    if not isinstance(logger, Logger):
-        raise TypeError("Logger should inherit logging.Logger class")
+    def _has_method(method_name):
+        return callable(getattr(logger, method_name, None))
```

Comment:
```suggestion
    def _has_method(logger: Any, method_name: str) -> bool:
        return getattr(logger, method_name, None) is not None
```

I'd expect a function called `_had_method()` to return a boolean, and I'd prefer if it didn't reach up into the calling environment.

Would you consider this format?

Reply:
oh! I just realized that `callable()` is checking whether or not that attribute obtained with `getattr()` exists and is callable! Sorry, that was my mistake reading this code.

I think just passing in `logger` and adding the `->` bool type hint is required.

```suggestion
    def _has_method(logger: Any, method_name: str) -> bool:
        return callable(getattr(logger, method_name, None))
```